src/NvFetcher/NixExpr.hs

```diff
@@ -77,6 +77,34 @@ nixFetcher sha256 = \case
             sha256 = $sha256;
           }
     |]
+  FetchGitHub
+    { _sha256 = _,
+      _rev = asString . toNixExpr -> rev,
+      _fetchSubmodules = toNixExpr -> fetchSubmodules,
+      _deepClone = toNixExpr -> deepClone,
+      _leaveDotGit = toNixExpr -> leaveDotGit,
+      _fowner = asString -> owner,
+      _frepo = asString -> repo
+    } ->
+      -- TODO: fix fetchFromGitHub in Nixpkgs so that deepClone and
+      -- leaveDotGit won't get passed to fetchzip
+      --
+      -- NOTE: the below code tries to determine which of fetchgit and
+      -- fetchzip would be used
+      [trimming|
+          fetchFromGitHub ({
+            owner = $owner;
+            repo = $repo;
+            rev = $rev;
+            fetchSubmodules = $fetchSubmodules;
+            sha256 = $sha256;
+          } // (
+            if ($fetchSubmodules || $leaveDotGit || $deepClone) then {
```

Comment:
This is the same conditional logic that is used in `fetchFromGitHub`:

https://github.com/NixOS/nixpkgs/blob/6eae8a116011f4db0aa5146f364820024411d6bb/pkgs/build-support/fetchgithub/default.nix#L13

Reply:
I think we can move the if logic to Haskell code and directly generate nix expr that calls `fetchgit` when necessary.

Reply:
I moved the conditional to Haskell code and also removed the check for `fetchSubmodules` because I realized it was unnecessary.

I had to resort to string comparisons though, because I couldn't grok enough of the Haskell syntax. Maybe there is a better way?

Reply:
Thanks, that's good enough. I can clean up later